src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -529,48 +556,85 @@ protected override void StopProcessing()
-            }
+                // Memory Stream
+                _ms = new MemoryStream();
+
+                // We use XmlTextWriter originally:
+                //     _xw = new XmlTextWriter(_ms, null);
+                //     _xw.Formatting = Formatting.Indented;
+                // This implies the following settings:
+                //  - Encoding is null -> use the default encoding 'UTF-8' when creating the writer from the stream;
+                //  - XmlTextWriter closes the underlying stream / writer when 'Close/Dispose' is called on it;
+                //  - Use the default indentation setting -- two space characters.
+                //
+                // We configure the same settings in XmlWriterSettings when refactoring this code to use XmlWriter:
+                //  - The default encoding used by XmlWriterSettings is 'UTF-8', but we call it out explicitly anyway;
+                //  - Set CloseOutput to true;
+                //  - Set Indent to true, and by default, IndentChars is two space characters.
+                //
+                // We use XmlWriterSettings.OmitXmlDeclaration instead of XmlWriter.WriteStartDocument because the
+                // xml writer created by calling XmlWriter.Create(Stream, XmlWriterSettings) will write out the xml
+                // declaration even without calling WriteStartDocument().
+
+                var xmlSettings = new XmlWriterSettings();
+                xmlSettings.Encoding = Encoding.UTF8;
+                xmlSettings.CloseOutput = true;
+                xmlSettings.Indent = true;
+
+                if (As.Equals("Stream", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Omit xml declaration in this case because we will write out the declaration string in BeginProcess.
+                    xmlSettings.OmitXmlDeclaration = true;
+                }
 
-            _xw = XmlWriter.Create(_ms, xmlSettings);
+                _xw = XmlWriter.Create(_ms, xmlSettings);
 
-            if (Depth == 0)
-            {
-                _serializer = new CustomSerialization(_xw, NoTypeInformation);
+                if (Depth == 0)
+                {
+                    _serializer = new CustomSerialization(_xw, NoTypeInformation);
+                }
+                else
+                {
+                    _serializer = new CustomSerialization(_xw, NoTypeInformation, Depth);
+                }
             }
             else
             {
-                _serializer = new CustomSerialization(_xw, NoTypeInformation, Depth);
+                // // create xml writer
+                _sw = new StringWriter();
+                XmlWriterSettings xmlSettings = new XmlWriterSettings();
+                xmlSettings.CloseOutput = true;
+                xmlSettings.Encoding = _sw.Encoding;
+                xmlSettings.Indent = true;
+                xmlSettings.OmitXmlDeclaration = true;
+                _xw = XmlWriter.Create(_sw, xmlSettings);
```

Comment:
This code & the block immediately above it look pretty similar. See if you can find a way to refactor it so that you're not repeating yourself quite so much. 🙂